ruddock/modules/users/routes.py

```diff
@@ -26,17 +27,64 @@ def show_memberlist(search_type):
       memberlist=memberlist,
       search_type=search_type)
 
-@blueprint.route('/view/<username>')
+@blueprint.route('/view/user_id=<user_id>')
 @login_required()
-def view_profile(username):
-  """Procedure to show a user's profile and membership details."""
-  user_info = helpers.get_user_info(username)
-  offices = helpers.get_office_info(username)
+def view_member(user_id):
+  """Procedure to show a member's profile and membership details if they have
+  no user account on the website."""
+  if user_id is not None:
+    member_info = helpers.get_member_info(user_id)
+    offices = helpers.get_office_info(user_id)
 
-  if user_info is not None:
-    return flask.render_template('view_user.html',
-        info=user_info,
-        offices=offices,
-        strftime=time.strftime)
-  else:
-    flask.abort(httplib.NOT_FOUND)
+    if member_info is not None:
+      return flask.render_template(
+          'view_user.html',
+          info=member_info,
+          offices=offices,
+          strftime=time.strftime,
+          edit_permission=auth_utils.check_permission(Permissions.USERS),
+          user_id=user_id)
+  flask.abort(httplib.NOT_FOUND)
+
+@blueprint.route('/manage/user_id=<user_id>', methods=['GET', 'POST'])
+@login_required(Permissions.USERS)
+def manage_member(user_id):
+  """Procedure to edit a member in the database if the user browsing the
+  website has the permissions necessary to edit users (USERS or ADMIN)."""
+  member_info = helpers.get_member_info(user_id)
+  if flask.request.method == 'GET':
+    if user_id is not None:
+      return flask.render_template(
+          'manage_member.html',
+          user_id=user_id,
+          info=member_info)
+  else: # POST request, so they're submitting a form to edit/remove a user.
+    if 'remove' in flask.request.form:
+      if helpers.remove_member(user_id):
+        flask.flash('Success!')
+      else:
+        flask.flash('An error occurred trying to perform your operation. '
+                    'Please contact an IMSS Rep!')
+    elif 'edit' in flask.request.form:
+      # Everything comes back from the web page as a unicode string, but
+      # when we input the database query, some of these need to be passed
+      # as numbers instead. Note that user_id is stored in the database as
+      # an int as well, but it's not part of the dictionary, so we just cast
+      # it manually when we call helpers.edit_member_column.
+      stored_as_int = ('member_type', 'msc', 'room_number')
+      success = True
+
+      # use dict.items() instead in Python 3
+      for key, value in flask.request.form.iteritems():
+        if value != '' and key != 'edit':
+          if key in stored_as_int:
+            value = int(value)
```

Comment:
you can actually pass just about anything into db.execute, and it'll do its best to interpret it. in particular, if you pass the string "939" as an msc, it will still be stored correctly as an int.